packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-sign-in-soc-idp/java/integrationsteps.md

```diff
@@ -1,3 +1,77 @@
 ## Integration steps
 
-The Java SDK integration steps for this use case are coming soon. See the [Okta Identity Engine Java SDK usage guide](https://github.com/okta/okta-idx-java#usage-guide) for more information.
+### Step 1: Start the sign-in process
+
+Use [IDXAuthenticationWrapper](https://github.com/okta/okta-idx-java/blob/master/api/src/main/java/com/okta/idx/sdk/api/client/IDXAuthenticationWrapper.java) to start the sign-in process with Okta when the user goes to the sign-in page.
+
+```java
+AuthenticationResponse beginResponse = idxAuthenticationWrapper.begin()
+```
+
+### Step 2: Get the available IdPs from your org
+
+Use the [AuthenticationResponse.getIdps()](https://github.com/okta/okta-idx-java/blob/master/api/src/main/java/com/okta/idx/sdk/api/response/AuthenticationResponse.java#L91) function to return a list of social Identity Provider (IdP) options configured in your org's routing rule. See [Set up your Okta org (for social Identity Providers)](/docs/guides/oie-embedded-common-org-setup/java/main/#set-up-your-okta-org-for-social-identity-providers) for IdP configuration in your org.
+
+```java
+ModelAndView modelAndView = new ModelAndView("login");
```

Comment:
Replace the entire code snippet content with:

```suggestion
List<Idp> idps = authenticationResponse.getIdps();
```

Reply:
Thanks! I will update this section along with a few more diagram and process-flow step edits.